src/ui/public/visualize/visualize.html

```diff
@@ -11,7 +11,7 @@ <h2 aria-hidden="true"><i aria-hidden="true" class="fa fa-meh-o"></i></h2>
     ng-style="loadingStyle"
     ng-class="{ loading: vis.type.requiresSearch && searchSource.activeFetchCount > 0 }"
     class="visualize-chart"></div>
-  <visualize-legend></visualize-legend>
+  <visualize-legend ng-show="vis.params.addLegend"></visualize-legend>
```

Comment:
Given that the `addLegend` option was remove from the **Options** tab on visualizations this will mean that any legacy visualization with `addLegend: false` will have no way of re-enabling the legend. I'm pretty sure this was only added for Discover. I'm ok with leaving the toggle in discover, let's remove this